waku/node/v2/waku_types.nim

```diff
@@ -144,6 +144,8 @@ type
     listenStr*: string
     #multiaddrStrings*: seq[string]
 
+  WakuResult*[T] = Result[T, cstring]
```

Comment:
Can you explain why this is needed? I don't understand what the cstring is doing here.

Reply:
Because I'm planning on only using string literals as error. Not adding any values that can change to the error string, so no need to use `string`.
I could have used `Result[T, cstring]` directly, but I created the type as I'm sure we can use it on other locations also.

Reply:
@kdeme I usually like the pattern of using an enum. That means that we can compare the error to an enum for error handling and can also print its string value. What do you think?

Reply:
Well sure, if there is a use case that such comparison is needed. If not, just keep the error string for logging, which is what I am doing now.

I'm fine with changing this later if there are such use cases.